.github/workflows/image-scan.yaml

```diff
@@ -0,0 +1,37 @@
+---
+# Copyright The Linux Foundation and each contributor to LFX.
+# SPDX-License-Identifier: MIT
+
+name: Image Scan
+
+"on":
+  pull_request: null
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  publish:
+    name: Image Scan
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+      - uses: ko-build/setup-ko@v0.8
+      - run: |
+          ko build . --push=false --tarball "${RUNNER_TEMP}/image.tar" \
+            --bare \
+            --platform linux/arm64 \
+            --sbom spdx
+      - name: Install trufflehog via shell pipe
+        run: |
+          curl -sSfL https://raw.githubusercontent.com/trufflesecurity/trufflehog/main/scripts/install.sh \
+            | sh -s -- -b /usr/local/bin
+      - run: |
+          trufflehog --github-actions filesystem "${RUNNER_TEMP}/image.tar"
```

Comment:
You probably want `--fail` here, which causes trufflehog to return with exit code 183 if any secrets are found.

Some secrets, like GH, also return a lot of unverified false positives. You can either do `--only-verified` to return only secrets that are verified.

You probably also want `--no-update` to stop trufflehog from trying to update itself.